test-refresh/src/com/jakemccrary/test_refresh.clj

```diff
@@ -163,13 +163,27 @@
 (defn- something-changed? [x y]
   (not= x y))
 
-(defn- monitor-keystrokes [keystroke-pressed]
+(def printing (atom false))
+
+(defn- monitor-keystrokes [keystroke-pressed with-repl?]
   (future
-    (loop [c (.read System/in)]
-      (if (= c -1)
-        (System/exit 0)
-        (do (reset! keystroke-pressed true)
-            (recur (.read System/in)))))))
+    (let [run-test-refresh! #(do (reset! keystroke-pressed true)
+                                 (reset! printing true))]
+      (if with-repl?
+        (clojure.main/repl
+          :prompt #(do (while @printing (Thread/sleep 100))
+                       (clojure.main/repl-prompt))
+          :read (fn [request-prompt request-exit]
+                  (if-let [line (read-line)]
+                    (if-not (empty? (clojure.string/trim line))
+                      (read-string line)
+                      (do (run-test-refresh!) :run-tests))
+                    (System/exit 0))))
+        (loop [line (read-line)]
```

Comment:
I realized now that this should have just stayed the same, I can change it back if you prefer.